File: 프로그래머스/lv3/12938. 최고의 집합/최고의 집합.py
def solution(n, s):
    if s<n:
        return [-1]
    answer = []
    a=s//n
    for i in range(n):
        answer.append(a)
    if s%n:
        s-=a*n
        for i in range(n):
            if not s:break
            answer[i]+=1
            s-=1
    answer.sort()
    return answer

File: 프로그래머스/lv3/12938. 최고의 집합/최고의 집합.cpp
#include <bits/stdc++.h>

using namespace std;
vector<int> solution(int n, int s) {
    vector<int> answer;
    if (n > s)
        return { -1 };
    int S = s;
    int N = n;
    int div = s / n;
    int num = 0;
    while (S)
    {
        if (n - answer.size() - (S / (div + 1)) == 0)
        {
            int si = answer.size();
            for (int i = 0; i < n - si; i++)
            {
                answer.push_back(div + 1);
            }
            break;
        }
        S -= div;
        answer.push_back(div);
    }
    sort(answer.begin(), answer.end());
    return answer;
}